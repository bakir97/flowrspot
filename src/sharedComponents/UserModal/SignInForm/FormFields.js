import React from "react";
import Input from "../../Input";
import styles from "./SignInForm.module.scss";
const FormFields = ({
  first_name,
  last_name,
  date_of_birth,
  email,
  password,
  onChange,
  modalType
}) => {
  const renderInput = (value, title, name, type, full) => (
    <Input
      title={title}
      value={value}
      name={name}
      onChange={onChange}
      type={type}
      full={full}
    />
  );
  if (modalType === "login") {
    return (
      <>
        {renderInput(email, "Email Address", "email", "email", true)}
        {renderInput(password, "Password", "password", "password", true)}
      </>
    );
  }
  return (
    <>
      <div className={styles.row_inputs_container}>
        {renderInput(first_name, "First Name", "first_name", "text")}
        {renderInput(last_name, "Last Name", "last_name", "text")}
      </div>
      {renderInput(
        date_of_birth,
        "Date of Birth",
        "date_of_birth",
        "date",
        true
      )}
      {renderInput(email, "Email Address", "email", "email", true)}
      {renderInput(password, "Password", "password", "password", true)}
    </>
  );
};

export default FormFields;
