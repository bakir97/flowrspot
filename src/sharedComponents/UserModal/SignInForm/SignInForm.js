import React, { Component } from "react";
import FormFields from "./FormFields";
import Modal from "../../Modal";
import SignInButton from "../../SignInButton";
import styles from "./SignInForm.module.scss";
export default class SignInForm extends Component {
  state = {
    first_name: "",
    last_name: "",
    date_of_birth: "",
    email: "",
    password: ""
  };
  componentDidUpdate(prevProps) {
    if (prevProps.modalType !== this.props.modalType) {
      this.props.errorUser([]);
      this.setState({
        first_name: "",
        last_name: "",
        date_of_birth: "",
        email: "",
        password: ""
      });
    }
  }
  onChange = e => {
    this.setState({ [e.target.name]: e.target.value });
  };

  onSubmit = e => {
    const { email, password } = this.state;
    e.preventDefault();
    if (this.props.modalType === "login") {
      return this.props.loginUser({ email, password });
    }
    this.props.registerUser(this.state);
  };
  renderErrors = () =>
    this.props.errors.map(errorMessage => (
      <p className={styles.error_text} key={errorMessage}>
        {errorMessage}
      </p>
    ));
  render() {
    const { errors, modalType } = this.props;
    const isLogin = modalType === "login" ? true : false;

    return (
      <Modal
        modalText={
          isLogin ? "I don’t want to Login" : "I don’t want to register"
        }
      >
        <form className={styles.form} onSubmit={this.onSubmit}>
          <p className={styles.title}>
            {isLogin ? "Welcome Back" : "Create an Account"}
          </p>
          <FormFields
            {...this.state}
            onChange={this.onChange}
            modalType={modalType}
          />

          {errors.length > 0 && this.renderErrors()}
          <SignInButton
            title={isLogin ? "Login to your Account" : "Create Account"}
            type="submit"
          />
        </form>
      </Modal>
    );
  }
}
