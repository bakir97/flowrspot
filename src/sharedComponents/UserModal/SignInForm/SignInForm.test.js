import React from "react";
import { shallow } from "enzyme";
import SignInForm from "./SignInForm";
describe("SignInForm", () => {
  it("should render correctly", () => {
    const component = shallow(<SignInForm errors={[]} />);
    expect(component).toMatchSnapshot();
  });
  it("should change state on input change", () => {
    const component = shallow(<SignInForm errors={[]} />);
    component
      .instance()
      .onChange({ target: { value: "test", name: "first_name" } });
    expect(component.state().first_name).toBe("test");
  });
});
