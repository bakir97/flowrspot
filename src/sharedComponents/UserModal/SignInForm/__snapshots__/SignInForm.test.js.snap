// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignInForm should render correctly 1`] = `
<Connect(Modal)
  modalText="I donâ€™t want to register"
>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <p
      className="title"
    >
      Create an Account
    </p>
    <FormFields
      date_of_birth=""
      email=""
      first_name=""
      last_name=""
      onChange={[Function]}
      password=""
    />
    <SignInButton
      title="Create Account"
      type="submit"
    />
  </form>
</Connect(Modal)>
`;
